tracer/src/Datadog.Tracer.Native/clr_helpers.cpp

```diff
@@ -231,83 +231,95 @@ TypeInfo GetTypeInfo(const ComPtr<IMetaDataImport2>& metadata_import, const mdTo
+        {
+            case mdtTypeDef:
+                hr = metadata_import->GetTypeDefProps(token, type_name, kNameMaxSize, &type_name_len, &type_flags,
+                                                      &type_extends);
 
-            type_isAbstract = IsTdAbstract(type_flags);
-            type_isSealed = IsTdSealed(type_flags);
+                metadata_import->GetNestedClassProps(token, &parent_type_token);
+                if (parent_type_token != mdTokenNil)
+                {
+                    parentTypeInfo = std::make_shared<TypeInfo>(GetTypeInfo(metadata_import, parent_type_token));
+                }
 
-            break;
-        case mdtTypeRef:
-            hr = metadata_import->GetTypeRefProps(token, &parent_token, type_name, kNameMaxSize, &type_name_len);
-            break;
-        case mdtTypeSpec:
-        {
-            PCCOR_SIGNATURE signature{};
-            ULONG signature_length{};
+                if (type_extends != mdTokenNil)
+                {
+                    extendsInfo = std::make_shared<TypeInfo>(GetTypeInfo(metadata_import, type_extends));
+                    type_valueType =
+                        extendsInfo->name == WStr("System.ValueType") || extendsInfo->name == WStr("System.Enum");
+                }
 
-            hr = metadata_import->GetTypeSpecFromToken(token, &signature, &signature_length);
+                type_isAbstract = IsTdAbstract(type_flags);
+                type_isSealed = IsTdSealed(type_flags);
 
-            if (FAILED(hr) || signature_length < 3)
+                break;
+            case mdtTypeRef:
+                hr = metadata_import->GetTypeRefProps(token, &parent_token, type_name, kNameMaxSize, &type_name_len);
+                break;
+            case mdtTypeSpec:
             {
-                return {};
-            }
+                PCCOR_SIGNATURE signature{};
+                ULONG signature_length{};
 
-            if (signature[0] & ELEMENT_TYPE_GENERICINST)
-            {
-                mdToken type_token;
-                CorSigUncompressToken(&signature[2], &type_token);
-                const auto baseType = GetTypeInfo(metadata_import, type_token);
-                return {baseType.id,        baseType.name,        token,
-                        token_type,         baseType.extend_from, baseType.valueType,
-                        baseType.isGeneric, baseType.isAbstract, baseType.isSealed,
-                        baseType.parent_type, baseType.scopeToken};
+                hr = metadata_import->GetTypeSpecFromToken(token, &signature, &signature_length);
+
+                if (FAILED(hr) || signature_length < 3)
+                {
+                    return {};
+                }
+
+                if (signature[0] & ELEMENT_TYPE_GENERICINST)
+                {
+                    mdToken type_token;
+                    CorSigUncompressToken(&signature[2], &type_token);
+                    token = type_token;
+                    continue;
```

Comment:
There's a behaviour change here. We already talked about it offline.

Reply:
Fixed in 961ba21351854df01f4e38631933082689d8a2ca